### PR TITLE
33 coding standard qml

### PR DESCRIPTION
**Pull request by Guitarheroua**

### CODE REVIEW

File: doc/workflow/coding_standard_qml.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+# Coding Standard QML
+
+###1. QML document structure
+
+#####Rule 1.1. Basic document and elements structure should be order like this:
```
Comment (alexchmykhalo) [inline doc/workflow/coding_standard_qml.md]: Grammar: "should be ordered" or you could even re-phrase whole setence: "Elements in the qml file should be ordered for each element in the following way:"

Comment (alexchmykhalo) [inline doc/workflow/coding_standard_qml.md]: I'd like to have some codes for the rules. Normally, it is very hard to distinguish between rules from diferrent sectionsa and very hard to remeber (though it is hard to remember every rule from coding standard anyway). My suggestion is to assign every rule the code that consists from shortened word and rule number. Example (for QML structure section): Struct1, Struct2, (for code formatting section): Format1, Format2, ...

File: doc/workflow/coding_standard_qml.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        
+        anchor.fill: parent
+        
+        onClicked: {
+            clicked()
+        }
+    }
+}
+```
+##### Rule 1.3. Grouped logically related properties.
```
Comment (alexchmykhalo) [inline doc/workflow/coding_standard_qml.md]: Please, re-phrase this rule. It is not clear what to do according to this.

File: doc/workflow/coding_standard_qml.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+}
+```
+##### Rule 1.5. All "magic" numbers and text names should be move to "readonly" properties
+```js
+Rectangle {
+   readonly property int meaningOfLife: 42
+   readonly property string stateName: "stateName"
+}
+```
+##### Rule 1.6. Each "*.qml" file should have only one top level (Root) element.
```
Comment (alexchmykhalo) [inline doc/workflow/coding_standard_qml.md]: Is it possible to have few top level elements in QML?

File: doc/workflow/coding_standard_qml.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    //Do thomething
+  } else {
+    //Do thomething else
+  }
+  
+  function myFunction (parameter) {
+    //Important calculation
+  }
+```
+#####Rule 3.2.2 Do not use semicolon if there is single or simple expression
```
Comment (alexchmykhalo) [inline doc/workflow/coding_standard_qml.md]: What you mean by "semicolon"? Do you mean curly brackets?

**Comment by alexchmykhalo** [comment]
Please, make changes according to code review comments.

**Comment by alexchmykhalo** [comment]
To avoid such a big number of automatic commits, pleaase use manual commits in Git book.
You could uncheck Setting->Git->Automatically commit changes, then use Save buton + Commit button (Left part of toolbar) to make a commit.

File: doc/workflow/coding_standard_qml.md (hunk 1 above)
Comment (alexchmykhalo) [inline doc/workflow/coding_standard_qml.md]: So what about my commet above (about rule codes)? Or you prefer to leave existing names for the rules?

Comment (Guitarheroua) [inline doc/workflow/coding_standard_qml.md]: Oh! Sorry, Didn't understand you correctly! Will fix that!

Comment (alexchmykhalo) [inline doc/workflow/coding_standard_qml.md]: I mean, that is a matter for discussion. But anyway any comment in code review should be followed by fix or discussion (thats why I did not approved request). We probaly need to add a new best practive for that in workflow.

**Comment by alexchmykhalo** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/PLLUG/pllug-presentation-system-2/pull-request/53/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=PLLUG/pllug-presentation-system-2)

**Comment by alexchmykhalo** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/PLLUG/pllug-presentation-system-2/pull-request/53/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=PLLUG/pllug-presentation-system-2)

**Pull request merged**